### PR TITLE
timeline: don't clear the internal counter in presence of local echoes

### PR DESCRIPTION
**Pull request by bnjbvr**
Might fix issues like https://github.com/element-hq/element-x-android-rageshakes/issues/2585 or items morphing into other items as in https://github.com/element-hq/element-x-ios/issues/3206.

### CODE REVIEW

File: crates/matrix-sdk-ui/src/timeline/tests/echo.rs [hunk 1]
```diff
@@ -208,16 +203,9 @@ async fn test_day_divider_duplication() {
...
-        .handle_live_event(sync_timeline_event!({
-            "content": {
-                "body": "B",
-                "msgtype": "m.text",
-            },
-            "sender": &*BOB,
-            "event_id": event_id,
-            "origin_server_ts": 1,
-            "type": "m.room.message",
-        }))
```
Comment (jplatte) [inline crates/matrix-sdk-ui/src/timeline/tests/echo.rs:220]:
```
thread 'timeline::echo::test_dedup_by_event_id_late' panicked at crates/matrix-sdk-ui/tests/integration/timeline/echo.rs:264:9:
assertion failed: `Set { index: 1, value: TimelineItem { kind: Event(EventTimelineItem { sender: "@example:localhost", sender_profile: Unavailable, reactions: ReactionsByKeyBySender({}), timestamp: 2024-08-29T16:12:09.374, content: Message(Message { in_reply_to: None, thread_root: None, edited: false, .. }), kind: Local(LocalEventTimelineItem { send_state: Sent { event_id: "$wWgymRfo7ri1uQx0NXO40vLJ" }, transaction_id: "b7c33097391e40f8837413a633b9b9e2", send_handle: Some(SendHandle { room: RoomSendQueue { .. }, transaction_id: "b7c33097391e40f8837413a633b9b9e2" }) }), is_room_encrypted: Some(false) }), internal_id: "0" } }` does not match `VectorDiff::PushFront { value }`
```

File: crates/matrix-sdk-ui/tests/integration/timeline/echo.rs [hunk 2]
```diff
@@ -253,18 +251,10 @@ async fn test_dedup_by_event_id_late() {
...
 
+    let f = EventFactory::new();
     sync_builder.add_joined_room(JoinedRoomBuilder::new(room_id).add_timeline_event(
-        sync_timeline_event!({
-            "content": {
-                "body": "Hello, World!",
-                "msgtype": "m.text",
-            },
-            "event_id": event_id,
-            "origin_server_ts": 123456,
```
[the remark this answers is not in the document]
Comment (bnjbvr) [reply crates/matrix-sdk-ui/tests/integration/timeline/echo.rs:263]: And I was wrong! it mostly needed a bigger timeout in the response mock, for some reason. Too bad wiremock doesn't allow for multiple queries to be handled at the same time, as it's really bad at having async responses…

[the remark this answers is not in the document]
Comment (jplatte) [reply crates/matrix-sdk-ui/tests/integration/timeline/echo.rs:263]: Oops, left comment in the wrong place. I even looked at both and remember concluding it was the integration test that failed.. 😅

**Review by Hywan — APPROVED** [review]
Looks sane to me.

**Pull request merged**